src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -4008,20 +4008,32 @@ public virtual IEnumerator<T> GetEnumerator()
     /// </summary>
     internal class PSMemberInfoInternalCollection<T> : PSMemberInfoCollection<T>, IEnumerable<T> where T : PSMemberInfo
     {
-        private readonly OrderedDictionary _members;
+        private OrderedDictionary _members;
         private int _countHidden;
 
+        private OrderedDictionary Members
+        {
+            get
+            {
+                if (_members == null)
+                {
+                    _members = new OrderedDictionary(StringComparer.OrdinalIgnoreCase);
+                }
```

Comment:
There could be a race condition here.
In an exaggerated way, thread A checked `_members == null` and entered the if block, then A got swapped out, and thread B came in and used `Members` (so `_member` is initialized). When A got swapped in again, it would re-create the `OrderedDictionary` object, and thus any members B added in will be gone.

Reply:
I suggest the following code:
```
if (_members == null)
{
    Interlocked.CompareExchange(ref _members, new OrderedDictionary(StringComparer.OrdinalIgnoreCase), null);
}

return _members;
```

Reply:
Great catch!

Fixed.